openedx/core/lib/gating/api.py

```diff
@@ -299,3 +304,112 @@ def get_gated_content(course, user):
                 {'id': user.id}
             )
         ]
+
+
+def is_prereq_met(content_id, user_id, recalc_on_unmet=False):
+    """
+    Returns true if the prequiste has been met for a given milestone
+
+    Arguments:
+        content_id (BlockUsageLocator): BlockUsageLocator for the content
+        user_id: The id of the user
+        recalc_on_unmet: Recalculate the grade if prereq has not yet been met
+
+    Returns:
+        tuple: True|False,
+        prereq_meta_info = { 'url': prereq_url|None, 'display_name': prereq_name|None}
+    """
+    course_id = content_id.course_key
+
+    # if unfullfilled milestones exist it means prereq has not been met
+    unfulfilled_milestones = milestones_helpers.get_course_content_milestones(
+        course_id,
+        content_id,
+        'requires',
+        user_id
+    )
+    prereq_met = not unfulfilled_milestones
+    prereq_meta_info = {'url': None, 'display_name': None}
+
+    if prereq_met or not recalc_on_unmet:
+        return prereq_met, prereq_meta_info
+
+    milestone = unfulfilled_milestones[0]
+    student = User.objects.get(id=user_id)
+    store = modulestore()
+
+    with store.bulk_operations(course_id):
+        course_structure = get_course_blocks(student, store.make_course_usage_key(course_id))
```

Comment:
You should be able to pass in just the `subsection_usage_key` here instead of the course' usage key.  This will then compute the course blocks for only this particular subsection's subtree.

Reply:
excellent, will try